src/engine/Engine.cpp

```diff
@@ -416,6 +358,132 @@ void Engine::mainLoopStatistics()
                                   TimeUtils::timePassed( start, end ) );
 }
 
+inline void Engine::performBoundTighteningAfterCaseSplit()
+{
+    // Tighten bounds of a first hidden layer with MILP solver
+    performMILPSolverBoundedTighteningForSingleLayer( 1 );
+    do
+    {
+        performSymbolicBoundTightening();
+    }
+    while ( applyAllValidConstraintCaseSplits() );
+
+    // Tighten bounds of an output layer with MILP solver
+    if ( _networkLevelReasoner )    // to avoid failing of system test.
+        performMILPSolverBoundedTighteningForSingleLayer
+            ( _networkLevelReasoner->getLayerIndexToLayer().size() - 1 );
+}
+
+
+inline bool Engine::handleSatisfyingAssignmentToConvexRelaxation()
```

Comment:
This method is called when a satisfying assignment is already found for the linear constraints. The next step is to attempt to satisfy the PLConstraints as well. I renamed it to handleSatisfyingAssignmentToLinearConstraints()

Reply:
I still think it could be a bit misleading, especially because the linear constraints are already satisfied. Perhaps "adjustSatisfyingAssigmentToNonLinearConstraints", or something similar? I know it's a bit cluttered...

Reply:
I see now how it can be confusing.. Probably adjustAssignmentToSatisfyNonLinearConstraints" is the least confusing. Will change to that.